### PR TITLE
Qt 5 fixes

### PR DESCRIPTION
**Pull request by jonaski**
This PR includes the following fixes. Most of them are to make it compile natively on macos and cross-compile with mingw using a linux machine for windows:

- Don't assume Linux is Unix
- Make it compile without X11 and with dbus (or the opposite), (macos has dbus but not X11. Some BSD's have X11 and not dbus).
- Replace qtwin with Qt5::WinExtras
- Remove broken json include directory from libmygpo-qt5
- Make Qocoa compile on macos
- Replace deprecated Q_WS macros with Q_OS (I'm using HAVE_X11 instead of Q_WS_X11)
- Remove unused X11 code from qtsingleapplication (dead code and it won't compile with Qt 5)
- Fix qtsingleapplication compile on windows (these changes are already in master)
- Make Spotify optional (1 less dependency)
- Make translations optional
- Replace QStyleOptionViewItemV2-4 (deprecated in Qt 5)
- Update debian control file (tested on Ubuntu 18 Bionic)
- Update spec file (Tested on Fedora 29)
- Make it possible to compile on macos and windows without hard-coding paths with USE_BUNDLE=OFF
- Fix macdevicelister compile
- Fix chromaprinter cross compile with mingw
- Fix tinysvcmdns cross compile with mingw

Cross-compiling the qt 5 branch on linux for windows now works using mingw and qt 5.11.2
On macos it is possible to compile natively using dependencies from brew for those who want to test:

```
brew install glib pkgconfig cmake boost protobuf protobuf-c qt cryptopp gettext
brew install sqlite --with-fts
brew install gstreamer gst-plugins-base
brew install gst-plugins-good --with-flac
brew install gst-plugins-bad gst-plugins-ugly
brew install chromaprint liblastfm
export Qt5_DIR=/usr/local/opt/qt5/lib/cmake
export Qt5LinguistTools_DIR=/usr/local/Cellar/qt/5.11.2/lib/cmake/Qt5LinguistTools
export GST_SCANNER_PATH=/usr/local/Cellar/gstreamer/1.14.4/libexec/gstreamer-1.0/gst-plugin-scanner
export GST_PLUGIN_PATH=/usr/local/lib/gstreamer-1.0
export PATH="/usr/local/opt/gettext/bin:$PATH"

git clone https://github.com/jonaski/Clementine
cd Clementine
git checkout qt5-fix
cd bin
cmake .. -DQT_MAC_USE_COCOA=ON -DENABLE_VISUALISATIONS=OFF -DUSE_BUNDLE=OFF
make -j8
```

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -838,16 +831,35 @@ optional_source(HAVE_LIBLASTFM
...
-optional_source(HAVE_SPOTIFY_DOWNLOADER
+# Spotify support
+optional_source(HAVE_SPOTIFY
   SOURCES
-    internet/spotify/spotifyblobdownloader.cpp
+    internet/spotify/spotifyserver.cpp
+    internet/spotify/spotifyservice.cpp
+    internet/spotify/spotifysettingspage.cpp
+    globalsearch/spotifysearchprovider.cpp
+    songinfo/spotifyimages.cpp
```
Comment (hatstand) [inline src/CMakeLists.txt]: This shouldn't be behind the Spotify flag - it's not really related to the other Spotify support.

File: src/songinfo/artistinfoview.cpp [hunk 2]
```diff
@@ -20,12 +20,16 @@
...
 #include "songinfo/songkickconcerts.h"
-#include "songinfo/spotifyimages.h"
 #include "widgets/prettyimageview.h"
+#ifdef HAVE_SPOTIFY
+#include "songinfo/spotifyimages.h"
+#endif
 
 ArtistInfoView::ArtistInfoView(QWidget* parent) : SongInfoBase(parent) {
   fetcher_->AddProvider(new SongkickConcerts);
+#ifdef HAVE_SPOTIFY
```
Comment (hatstand) [inline src/songinfo/artistinfoview.cpp]: Again, this shoudn't be behind a flag.

**Comment by tomachinz** [comment]
Does anything below highlight my issue? Seems maybe I should re-install using "with-default-names" perhaps? 

So I did a brew install make and saw...

```
==> Downloading https://homebrew.bintray.com/bottles/make-4.2.1_1.high_sierra.bottle.1.tar.gz
######################################################################## 100.0%
==> Pouring make-4.2.1_1.high_sierra.bottle.1.tar.gz
==> Caveats
All commands have been installed with the prefix 'g'.
If you do not want the prefix, install using the "with-default-names" option.

If you need to use these commands with their normal names, you
can add a "gnubin" directory to your PATH from your bashrc like:

    PATH="/usr/local/opt/make/libexec/gnubin:$PATH"


➜  gnubin gmake -v
GNU Make 4.2.1
Built for x86_64-apple-darwin17.7.0
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
➜  gnubin make -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
➜  gnubin cmake -v
CMake Error: The source directory "/usr/local/opt/make/libexec/gnubin/-v" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
```

**Comment by hatstand** [comment]
Awesome. I think the biggest blocker for us moving to Qt5 is having reliable CI. Unfortunately cross compiling Qt5 is hard but if you're interested in getting it working, the docker image we currently use is here: https://github.com/clementine-player/Dependencies/blob/master/macosx/Dockerfile

**Comment by hatstand** [comment]
Yeah, that might make sense now that it's supported on travis. The problem then would be how do we lock dependency versions. We don't want to have to keep dealing with breakages caused by random package version changes.

**Comment by hatstand** [comment]
I've had some success using the binary bottles from brew instead of compiling from source so that might be a good direction.

**Comment by JulianVolodia** [comment]
I think choosing osx_image: xcode9.2 with Homebrew will be best option for now.

https://docs.travis-ci.com/user/reference/osx
https://docs.travis-ci.com/user/installing-dependencies/#using-a-brewfile

https://docs.travis-ci.com/user/reference/osx
https://doc.qt.io/qt-5/macos.html

**Pull request merged**